pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +106,243 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// update in place
+	updatedMapiMachineSet := mapiMachineSet.DeepCopy()
+
+	// Update the MAPI MachineSet spec
+	updateMAPIMachineSetSpecFromCAPI(ctx, updatedMapiMachineSet, capiMachineSet,
+		convertedMachineSet.Spec.Template.Spec.ProviderSpec)
+
+	// Check if there are any changes after updating the spec
+	if !reflect.DeepEqual(updatedMapiMachineSet.Spec, mapiMachineSet.Spec) {
+		logger.Info("Updating MAPI MachineSet spec")
+
+		if err := r.Update(ctx, updatedMapiMachineSet); err != nil {
+			logger.Error(err, "Failed to update MAPI MachineSet")
+
+			r.updateSynchronisedConditionWithPatch(
+				ctx, mapiMachineSet, corev1.ConditionFalse,
+				"FailedToUpdateMAPI", err.Error(),
+			)
+
+			return ctrl.Result{}, fmt.Errorf("failed to update MAPI MachineSet: %w", err)
+		}
+
+		logger.Info("Successfully updated MAPI MachineSet spec")
+	} else {
+		logger.Info("No changes detected in MAPI MachineSet spec")
+	}
+
+	// Update the MAPI MachineSet status
+	updateMAPIMachineSetStatusFromCAPI(ctx, updatedMapiMachineSet, capiMachineSet)
+
+	// Check if there are any changes after updating the status
+	if !reflect.DeepEqual(updatedMapiMachineSet.Status, mapiMachineSet.Status) {
+		logger.Info("Updating MAPI MachineSet status")
+
+		if err := r.Status().Update(ctx, updatedMapiMachineSet); err != nil {
+			logger.Error(err, "Failed to update MAPI MachineSet status")
+
+			r.updateSynchronisedConditionWithPatch(
+				ctx, mapiMachineSet, corev1.ConditionFalse,
+				"FailedToUpdateMAPI", err.Error(),
+			)
+
+			return ctrl.Result{}, fmt.Errorf("failed to update MAPI MachineSet status: %w", err)
+		}
+
+		logger.Info("Successfully updated MAPI MachineSet status")
+	} else {
+		logger.Info("No changes detected in MAPI MachineSet status")
+	}
+
+	r.updateSynchronisedConditionWithPatch(ctx, mapiMachineSet, corev1.ConditionTrue,
+		consts.ReasonResourceSynchronised, "Synchronised CAPI to MAPI")
+
 	return ctrl.Result{}, nil
 }
 
+func (r *MachineSetSyncReconciler) updateSynchronisedConditionWithPatch(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, status corev1.ConditionStatus, reason, message string) {
+	logger := log.FromContext(ctx).WithValues("function", "updateSynchronisedConditionWithPatch")
+
+	oldMachineSet := mapiMachineSet.DeepCopy()
+
+	var newCondition *machinev1beta1.Condition
+	if status == corev1.ConditionTrue {
+		newCondition = conditions.TrueConditionWithReason(
+			consts.SynchronisedCondition, reason, message, //nolint:govet
+		)
+
+		// Update the synchronizedGeneration
+		mapiMachineSet.Status.SynchronizedGeneration++
+	} else {
+		newCondition = conditions.FalseCondition(
+			consts.SynchronisedCondition, reason,
+			machinev1beta1.ConditionSeverityError, message, //nolint:govet
+		)
+	}
+
+	conditions.Set(&mapiMachineSet.Status.Conditions, newCondition)
+
+	patch := client.MergeFrom(oldMachineSet)
+	if err := r.Status().Patch(ctx, mapiMachineSet, patch); err != nil {
```

Comment:
I think controller runtime has an apply style patch option, that we should be using here, to set the correct field owner for this particular condition